### PR TITLE
E2E & features from initial plugin version

### PR DESCRIPTION
**Pull request by michaelfmnk**

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -46,14 +49,23 @@ func NewClient(baseUrl, clientId, clientSecret string) (*Client, error) {
...
+			Timeout: 30 * time.Second,
+		},
+		BaseURL:     baseUrl,
+		Token:       credentials.AccessToken,
+		RateLimiter: rate.NewLimiter(rate.Every(1*time.Second), 5),
 	}, nil
 }
 
 func (c *Client) DoRequest(req *http.Request) ([]byte, error) {
+	err := c.RateLimiter.Wait(context.Background())
```
Comment (vladimir-kozyrev) [inline client/client.go:64]: Finally! We will have rate limiting. Thanks. :)
It will probably fix #11.

File: client/connector.go [hunk 2]
```diff
@@ -114,3 +123,16 @@ func (c *Client) DeleteConnector(connectorId string, networkItemId string, netwo
 	_, err = c.DoRequest(req)
 	return err
 }
+
+func (c *Client) GetConnectorProfile(id string) (string, error) {
+	req, err := http.NewRequest(http.MethodPost, fmt.Sprintf("%s/api/beta/connectors/%s/profile", c.BaseURL, id), bytes.NewBufferString(""))
```
Comment (vladimir-kozyrev) [inline client/connector.go]: I guess you can just use `nil` instead of `bytes.NewBufferString("")` if the request body is not required.

**Comment by vladimir-kozyrev** [comment]
@michaelfmnk thank you for the PR.
Do you mind resolving the `go.mod` and `go.sum` conflicts?

File: client/client.go (hunk 1 above)
Comment (michaelfmnk) [reply client/client.go:64]: hopefully 🤞

**Comment by michaelfmnk** [comment]
> @michaelfmnk thank you for the PR. Do you mind resolving the `go.mod` and `go.sum` conflicts?

done

File: client/connector.go (hunk 2 above)
Comment (michaelfmnk) [reply client/connector.go]: replaced it with nil

**Review by vladimir-kozyrev — APPROVED** [review]
lgtm ✔️

**Pull request merged**